src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Unix.cs

```diff
@@ -2,62 +2,133 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.Win32.SafeHandles;
 using System;
 using System.Diagnostics;
+using System.Runtime.InteropServices;
 using System.Threading;
 
 namespace Internal.Runtime.Augments
 {
     public sealed partial class RuntimeThread
     {
-        private readonly WaitSubsystem.ThreadWaitInfo _waitInfo;
+        private static bool s_setThreadExitCallback;
 
-        private void PlatformSpecificInitialize() { }
+        // Event signaling that the thread has stopped
+        private ManualResetEvent _stopped;
 
-        private void PlatformSpecificInitializeExistingThread() { }
+        private readonly WaitSubsystem.ThreadWaitInfo _waitInfo;
 
         internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
 
-        private static ThreadPriority GetPriority() { return ThreadPriority.Normal; }
+        private void PlatformSpecificInitialize()
+        {
+            // Race condition is OK here since we set to the same value
+            if (!s_setThreadExitCallback)
+            {
+                Interop.Sys.RuntimeThread_SetThreadExitCallback(AddrofIntrinsics.AddrOf<Action>(OnThreadExit));
+                s_setThreadExitCallback = true;
+            }
+        }
 
-        private static bool SetPriority(ThreadPriority priority) { return true; }
+        private void PlatformSpecificInitializeExistingThread()
+        {
+            _stopped = new ManualResetEvent(false);
+        }
 
-        private bool HasFinishedExecution()
+        /// <summary>
+        /// Returns true if the thread is started or being started in StartInternal.
+        /// </summary>
+        private bool HasStarted()
         {
-            // TODO: Return true if the thread has finished execution
-            return false;
+            return _stopped != null;
         }
 
-        private bool JoinCore(int millisecondsTimeout)
+        private ThreadPriority GetPriorityLive()
         {
-            // TODO: Join the thread
-            return false;
+            return ThreadPriority.Normal;
         }
 
-        private void StartCore(object parameter)
+        private bool SetPriorityLive(ThreadPriority priority)
         {
-            // TODO: Start a new thread using _threadStart and parameter
+            return true;
         }
 
-        public ApartmentState GetApartmentState()
+        [NativeCallable]
+        private static void OnThreadExit()
```

Comment:
I'm not sure what else we can do here given that Wait Subsystem for Unix lives in managed code in CorLib. The state of the managed world should still be intact at this point but we do need to double check that GC can safely happen here (for example, by inducing a GC in this function).